README.md

```diff
@@ -1,72 +1,298 @@
+
+- A lock file usually placed at `/opt/puppetlabs/pe_event_forwarding/cache/state/events_collection_run.lock`. This setting is based on the `$settings::statedir` such that if the value for `statedir` is moved, the lockfile will move with it.
+
+- A directory for resource files based on the `$settings::confdir` variable. Usually this will be `/etc/puppetlabs/pe_event_forwarding`.
+
+Inside that directory will be:
+
+- The events collection script `collect_api_events.rb`.
+
+- A configuration settings file `events_collection.yaml`.
+
+- An index tracking file `pe_event_forwarding_indexes.yaml`.
+
+- A sub directory for ruby class files `api`.
+
+- A sub directory for utilities files consumed by the classes `util`
+
+- A `processors.d` directory where third party modules will place Event Forwarding processors. Modules are allowed to create sub directories to store library files to help the processors.
+
+## Logging
+
+This module will handle logging it's own messages, as well as the messages emitted by the processors it invokes.
+
+By default the log file will be placed in `/var/log/puppetlabs/pe_event_forwarding/pe_event_forwarding.log`. This path is dependenant on either the the value of `$settings::logdir`, or the `logdir` parameter for the `pe_event_forwarding` class. See below for details.
+
+The log file is in single line json format, except for the header at the top of the file. This means that the log can be parsed line by line, and each line will be a fully formed JSON object that can be parsed for processing.
+
+The default log level is `WARN`.
+
+## Advanced Configuration Options
+
+`pe_console`
+
+The PE Event Forwarding module is capable of running on any Linux machine in your estate. However, it assumes by default that it has been classified on the primary Puppet Server, and it will attempt to connect to the Puppet APIs on `localhost`. If you want to run event collection from another machine, you will have to set this variable to the host name of the machine running the PE APIs. If your PE instance is behind a load balancer or some other more complex configuration, set this variable to the DNS host name you would use for getting to the console in your browser.
+
+--------------------------------------------------------------------------------
+
+`disabled`
+
+Setting this variable to true will events collection from executing. No further events will gathered or sent by any processor until the parameter is removed or set to `true`. It will do this by removing the cron job that executes the events collection script. This is intended for use during maintenance windows of either PE or the recieving platforms, or for any other reason an administrator might want to stop sending events. Simply removing this class from a node configuration would not be enough to stop sending events.
```

Comment:
Add word 'prevent' after 'Setting this variable to true will'.

Reply:
'receiving' spelled wrong.

Reply:
Good catch.